nilearn/glm/first_level/first_level.py

```diff
@@ -1066,6 +1069,42 @@ def _prepare_mask(self, run_img):
                 self.masker_ = self.mask_img
 
 
+def _check_compatibility_mask_and_images(mask_img, run_imgs):
+    """Check that mask type and image types are compatible.
+
+    Images to fit should be a Niimg-Like
+    if the mask is a NiftiImage, NiftiMasker or a path.
+    Similarly, only SurfaceImages can be fitted
+    with a SurfaceImage or a SrufaceMasked as mask.
+    """
+    if mask_img is None:
+        return None
+
+    msg = (
+        "Mask and images to fit must be of compatible types.\n"
+        f"Got mask of type: {type(mask_img)}, "
+        f"and images of type: {[type(x) for x in run_imgs]}"
+    )
+
+    volumetric_type = (Nifti1Image, NiftiMasker, str, Path)
+    if isinstance(mask_img, volumetric_type) and any(
+        not isinstance(x, (Nifti1Image, str, Path)) for x in run_imgs
+    ):
+        raise TypeError(
+            f"{msg} "
+            f"where images should be NiftiImage-like instances "
+            f"(Nifti1Image or str or Path)."
+        )
+
+    surface_type = (SurfaceImage, SurfaceMasker)
```

Comment:
So we decide that str and Path are necessarily volumetric_type and surface_type ?

Reply:
Well at the moment, I don't think we have tackled surface file formats that contains both meshes and data (cifti is one them, right?), so we are always loading meshes and data from different 'sources' hence why I made the assumption that path must (FOR NOW) only be volumetric.

Does this make sense? I realize that I need to make that explicit somewhere, but not sure where (in which doc string? definitely in the user guide when we update it)